lib/install/management/store_files.go

```diff
@@ -189,3 +191,46 @@ func (d *Dispatcher) createVolumeStores(conf *metadata.VirtualContainerHostConfi
 	}
 	return nil
 }
+
+func (d *Dispatcher) deleteVolumeStoreIfForced(conf *metadata.VirtualContainerHostConfigSpec) {
+	if !d.force {
+		volumeStores := new(bytes.Buffer)
+		for label, url := range conf.VolumeLocations {
+			volumeStores.WriteString(fmt.Sprintf("\t%s: %s\n", label, url.Path))
+		}
+		log.Warnf("Since --force was not specified, the following volume stores will not be removed. Use the vSphere UI to delete content you do not wish to keep.\n%s", volumeStores.String())
+		return
+	}
+
+	for label, url := range conf.VolumeLocations {
+		// FIXME: url is being encoded by the portlayer incorrectly, so we have to convert url.Path to the right url.URL object
+		log.Debugf("Provided datastore URL %s", url.Path)
+		dsURL, err := vsphere.DatastoreToURL(url.Path)
+		log.Debugf("Parsed volume store path %s", dsURL.Path)
+		if err != nil {
+			log.Warnf("Didn't receive an expected volume store path format: %s", url.Path)
+		}
+		log.Infof("Deleting volume store %s on Datastore %s at path %s", label, dsURL.Host, dsURL.Path)
+
+		datastores, err := d.session.Finder.DatastoreList(d.ctx, dsURL.Host)
+
+		if err != nil {
+			log.Errorf("Error finding datastore %s: %s", dsURL.Host, err)
+			continue
+		}
+		if len(datastores) != 1 {
+			foundDatastores := new(bytes.Buffer)
+			for _, d := range datastores {
+				foundDatastores.WriteString(fmt.Sprintf("\n%s\n", d))
```

Comment:
Well, the slice returned by `DatastoreList` is a `[]object.Datastore*` so actually I found a bug here in that I was printing `d` and I wanted to print `d.InventoryPath` but in any case since I'm printing a member of an object and `datastores` is not a `[]string` I cannot use `strings.Join()`. Similar problem on line 202-204. 

There might be a slightly more concise way to express this, but the way I have it is straightforward, easy to understand, and fast.